pkg/operator/ceph/object/notification/controller.go

```diff
@@ -0,0 +1,251 @@
+	// workaround because the rook logging mechanism is not compatible with the controller-runtime logging interface
+	reconcileResponse, err := r.reconcile(request)
+	if err != nil {
+		logger.Errorf("failed to reconcile %v", err)
+	}
+
+	return reconcileResponse, err
+}
+
+func (r *ReconcileNotifications) reconcile(request reconcile.Request) (reconcile.Result, error) {
+	// fetch the CephBucketNotification instance
+	cephBN := &cephv1.CephBucketNotification{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, cephBN)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			logger.Debugf("CephBucketNotification %q resource not found. Ignoring since resource must be deleted.", request.NamespacedName)
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, errors.Wrapf(err, "failed to retrieve CephBucketNotification %q", request.NamespacedName)
+	}
+
+	// find the namespace for the ceph cluster (may be different than the namespace of the topic CR)
+	// Make sure a CephCluster is present otherwise do nothing
+	cephCluster, isReadyToReconcile, cephClusterExists, reconcileResponse := opcontroller.IsReadyToReconcileNoName(
+		r.client,
+		r.context,
+		controllerName,
+	)
+	if !isReadyToReconcile {
+		// This handles the case where the Ceph Cluster is gone and we want to delete that CR
+		if !cephBN.GetDeletionTimestamp().IsZero() && !cephClusterExists {
+			// Return and do not requeue. Successful deletion.
+			return reconcile.Result{}, nil
+		}
+		logger.Debug("Ceph cluster not yet present.")
+		return reconcileResponse, nil
+	}
+	r.clusterSpec = &cephCluster.Spec
+
+	// DELETE: the CR was deleted
+	if !cephBN.GetDeletionTimestamp().IsZero() {
+		logger.Debugf("CephBucketNotification %q was deleted", cephBN.Name)
+
+		// Return and do not requeue. Successful deletion.
+		return reconcile.Result{}, nil
+	}
+
+	// Populate clusterInfo during each reconcile
+	r.clusterInfo, _, _, err = mon.LoadClusterInfo(r.context, cephCluster.Namespace)
+	if err != nil {
+		return reconcile.Result{}, errors.Wrap(err, "failed to populate cluster info")
+	}
+
+	// create the notification if needed
+	return r.updateCephBucketNotification(cephBN)
+}
+
+func getCephObjectStore(cephOB bktv1alpha1.ObjectBucket) (objectStoreName types.NamespacedName, err error) {
+	// prase the following string: <namespace>-rgw-<store>.<namespace>.svc
+	// to ge the object store name and namespace
+	parsedBucketHost := strings.Split(cephOB.Spec.Endpoint.BucketHost, ".")
+	if len(parsedBucketHost) < 3 {
+		err = errors.Errorf("malformed BucketHost %q", cephOB.Spec.Endpoint.BucketHost)
+		return
+	}
+	parsedSubdomain := strings.Split(parsedBucketHost[0], "-rgw-")
+	if len(parsedSubdomain) != 2 {
+		err = errors.Errorf("malformed BucketHost subdomain %q", parsedBucketHost[0])
+		return
+	}
+	objectStoreName.Namespace = parsedSubdomain[0]
```

Comment:
Can't we use the same technique to infer the CephCluster since both CephCluster and CephObjectStore always share the same namespace?

Reply:
Yes, but if you can find the namespace of the objectstore, then you have the namespace of the cephcluster so you don't need `IsReadyToReconcileNoName()` right? YOu can just pass that namespace to `IsReadyToReconcile`.